src/ConstAddressDeployer.sol

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.9;
+
+contract ConstAddressDeployer {
+    event Deployed(bytes32 bytecodeHash, bytes32 salt, address deployedAddress);
+    function deploy(bytes memory bytecode, bytes32 salt) external returns (address deployedAddress) {
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            deployedAddress := create2(0, add(bytecode, 32), mload(bytecode), salt)
```

Comment:
Just to confirm, `add(bytecode, 32)` to skip its length?

Reply:
yes, the next argument (mload(bytecode)) is the length.